app/models/concerns/document.rb

```diff
@@ -4,11 +4,25 @@
 
 module Document
   extend ActiveSupport::Concern
+  IMMUNISATION_RECORD = 'ImmunisationRecord'
+  BIRTH_RECORD = 'BirthRecord'
+  DOCUMENT_TYPES = [IMMUNISATION_RECORD, BIRTH_RECORD].freeze
 
   included do # rubocop:disable Metrics/BlockLength
-    has_many :user_documents, dependent: :nullify
+    has_many :user_documents, dependent: :nullify, as: :document
     has_many :users, -> { distinct }, through: :user_documents
-    has_many :shares, -> { merge(Share.kept) }, dependent: :nullify, inverse_of: :document, foreign_key: 'document_id'
+    has_many :shares, -> { merge(Share.kept) }, as: :document, dependent: :nullify,
+                                                inverse_of: :document, foreign_key: 'document_id'
+
+    delegate :share_audit_comment, :remove_audit_comment, :add_audit_comment, :view_audit_comment, to: :class
+
+    def immunisation_record?
+      document_type == IMMUNISATION_RECORD
```

Comment:
this line isn't covered in the tests.  not used?

Reply:
hmm, should be used in `shares/index`